packages/kbn-ui-shared-deps-src/src/polyfills.js

```diff
@@ -17,3 +17,67 @@ if (typeof window.Event === 'object') {
 
 require('whatwg-fetch');
 require('symbol-observable');
+
+/* @notice
+ *
+ * Adapted from remote-web-worker, which was available under a "MIT" license.
+ *
+ * MIT License (MIT)
+ *
+ * Copyright (c) 2022 Jan Nicklas
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+typeof window !== 'undefined' &&
+  // eslint-disable-next-line no-global-assign
+  (Worker = ((BaseWorker) =>
+    class Worker extends BaseWorker {
+      constructor(scriptURL, options) {
+        const url = String(scriptURL);
```

Comment:
Questions/Nit: 
- Would we want to limit the remote URLs to http/https?
- `startsWith` might not catch a domain difference of `.co` vs `.com`

What are your thoughts on something like:
```js
const scriptUrl = new URL(url) // should work for `blob:` and `data:` urls as well.
const isCrossOrigin = (scriptUrl.protocol === 'https:' || scriptUrl.protocol === 'http:') && scriptUrl.origin !== window.location.origin;
```

Reply:
I think it'd make sense to limit remote URL to https, especially given that this patches every instantiation of workers across the project, and you're right  `startsWith` would not catch that subtle difference in the TLD that are similar to the example you provided. I adopted your suggestion with a slight modification, thanks @kc13greiner